builtin/mainmenu/tab_online.lua

```diff
@@ -331,6 +341,14 @@ local function main_button_handler(tabview, fields, name, tabdata)
 		end
 	end
 
+	if fields.btn_add_favorite then
+		local server = find_selected_server()
+
+		serverlistmgr.add_favorite(server)
+		set_selected_server(server)
```

Comment:
This shouldn't be necessary. Is it?

Reply:
I think it is, removing it won't update the selection after the server is marked as favorite.

Reply:
> I think it is, removing it won't update the selection after the server is marked as favorite.

But you can only mark an already selected server as favorite?

In other words: If `find_selected_server` returns something that is not `nil`, `server` is already selected and `set_selected_server(server)` should be unnecessary.

Reply:
Update after discussion on Discord: Turns out it is necessary for the side effect of updating `gamedata.fav`. Frankly we should just remove `gamedata.fav` eventually (it can be derived from the selected server) but for now this PR should simply set `gamedata.fav = true` here instead.